packages/ERTP/src/typeGuards.js

```diff
@@ -7,12 +7,19 @@ import { M, matches, getInterfaceGuardPayload } from '@endo/patterns';
  */
 
 export const BrandShape = M.remotable('Brand');
+harden(BrandShape);
```

Comment:
Hi @turadg , the extra `harden` should only appear when the declared variable is not already obviously initialized to a hardened value. For all `M.something(...)`, we should assume that it produces a properly hardened value and omit the noise.

We should usually only harden values produced by literal expressions right where the literal expression appears, ideally before the value of the expression might be observed in an unhardened state. For function literals we've done this immediately after the declarations. For object literals, I've generally done it inline before, but doing it after as you suggested is fine.

For values produced by APIs we assume are well behaved, like `M`, we should normally assume they've been hardened and avoid the noise.

```suggestion
export const BrandShape = M.remotable('Brand');
```

Reply:
That's a current limitation of the lint rule. I've filed https://github.com/endojs/endo/issues/2632

Until that's done, we have a tradeoff between the noise and missing some hardens. I'll leave to you to decide whether to keep the rule. If you prefer to continue without it feel free to force push back with your local branch. (Goes without saying as it's your PR 😄 )

Reply:
I did revert, but then I borrowed various improvements that came from your variant.

Reply:
With the additional clarification at https://github.com/endojs/endo/issues/2632#issuecomment-2477602697 , which this PR now follows, I've resolving this conversation.